src/main/java/software/amazon/payloadoffloading/S3BackedPayloadStore.java

```diff
@@ -13,16 +13,22 @@
 
     private final String s3BucketName;
     private final S3Dao s3Dao;
+    private final String s3Key;
 
     public S3BackedPayloadStore(S3Dao s3Dao, String s3BucketName) {
         this.s3BucketName = s3BucketName;
         this.s3Dao = s3Dao;
+        this.s3Key = UUID.randomUUID().toString();
+    }
+
+    public S3BackedPayloadStore(S3Dao s3Dao, String s3BucketName, String s3Key) {
+        this.s3BucketName = s3BucketName;
+        this.s3Dao = s3Dao;
+        this.s3Key = s3Key;
     }
 
     @Override
     public String storeOriginalPayload(String payload) {
-        String s3Key = UUID.randomUUID().toString();
```

Comment:
The s3 key is generated for each call to storePayload, in your case, you've changed it to be generated once in the constructor which would end up using the same s3 key each time, changing the default behaviour. I think instead of a second constructor, having a storeOriginalPayload(String payload, String key) that checks 
`String s3Key = (key == null) ? UUID.randomUUID().toString() : key`;
and have storeOriginalPayload(String payload) call storeOriginalPayload(payload, null)
The other libraries can then make use of this storeOriginalPayload(String payload, String key) if they want to provide such functionality, if not, it won't affect them.